lib/ansible/modules/cloud/packet/packet_device.py

```diff
@@ -477,46 +520,66 @@ def act_on_devices(target_state, module, packet_conn):
 
         state_map = {
             'absent': _absent_state_map,
-            'active': {'inactive': packet.Device.power_on},
+            'active': {'inactive': packet.Device.power_on,
+                       'provisioning': None, 'rebooting': None
+                       },
             'inactive': {'active': packet.Device.power_off},
             'rebooted': {'active': packet.Device.reboot,
-                       'inactive': packet.Device.power_on},
-            }
+                         'inactive': packet.Device.power_on,
+                         'provisioning': None, 'rebooting': None
+                         },
+        }
 
         # First do non-creation actions, it might be faster
         for d in process_devices:
+            if d.state == target_state:
+                continue
             if d.state in state_map[target_state]:
                 api_operation = state_map[target_state].get(d.state)
-                try:
+                if api_operation is not None:
                     api_operation(d)
                     changed = True
-                except Exception as e:
-                    _msg = ("while trying to make device %s, id %s %s, from state %s, "
-                            "with api call by %s got error: %s" %
-                           (d.hostname, d.id, target_state, d.state, api_operation, e))
-                    raise Exception(_msg)
             else:
-                _msg = ("I don't know how to process existing device %s from state %s "
-                        "to state %s" % (d.hostname, d.state, target_state))
+                _msg = (
+                    "I don't know how to process existing device %s from state %s "
+                    "to state %s" %
+                    (d.hostname, d.state, target_state))
                 raise Exception(_msg)
 
     # At last create missing devices
     created_devices = []
     if create_hostnames:
         created_devices = [create_single_device(module, packet_conn, n)
                            for n in create_hostnames]
-        if module.params.get('wait'):
-            created_devices = wait_for_ips(module, packet_conn, created_devices)
+        if module.params.get('wait_for_public_IPv'):
+            created_devices = wait_for_public_IPv(
+                module, packet_conn, created_devices)
         changed = True
 
     processed_devices = created_devices + process_devices
+    if target_state == 'active':
+        processed_devices = wait_for_devices_active(
+            module, packet_conn, processed_devices)
 
     return {
         'changed': changed,
         'devices': [serialize_device(d) for d in processed_devices]
     }
 
 
+def dig_error_json(http_err):
```

Comment:
Oh, this method seems a workaround of a bug in the Packet library. Indeed the library [already search](https://github.com/packethost/packet-python/blob/master/packet/baseapi.py#L83) for `error` key in response and use it to raise an exception but in fact this line is never reached because `raise_for_status` is [called before](https://github.com/packethost/packet-python/blob/master/packet/baseapi.py#L67). `ok` property [returns `False`](https://github.com/requests/requests/blob/master/requests/models.py#L687) only when `raise_for_status` raise an exception.